teps/0089-nonfalsifiable-provenance-support.md

```diff
@@ -224,24 +291,95 @@ The entrypointer image should also be able to see that this flag is set and acco
+tampering of the TaskRun status. Upon each retrieval of the TaskRun, the TektonController checks if the status is initialized,
+that the signature in the annotations validate the current status.
 The signature and SVID will be stored as annotations on the TaskRun, and can be verified by a client.
 
 ### Verification in Chains
-Once Signed Results are available, we'll add verification of Signed Results to Chains.
+Once Signed TaksRuns are available, we'll add verification of Signed TaskRun to Chains.
 If verification fails for a TaskRun then Chains will not sign it.
 
-
 ## Test Plan
 Tests for Signed Results:
 1. Enabling the alpha feature for SPIRE in Tekton
-1. Requesting an SVID & signature over Results for a TaskRun
-1. Verification of SPIRE with Chains
-1. Verify that a TaskRun that isn't created by Tekton isn't signed Chains
+1. Verify that a necessary fields are present in the pod status result
+1. Verify that a TaskRun pod status being modified with incorrect results isn't verified by the TektonContoller
 
 Tests for Signed TaskRuns:
-1. Verify that a TaskRun that's been modified during execution isn't verified
-1. Verify that a TaskRun that's been modified after execution isn't verified
-
+1. Verify that a TaskRun that's been modified during execution isn't verified by the TektonController
+1. Verify that a TaskRun that's been modified after execution isn't verified by the TektonController
+1. Verify that a TaskRun that's been modified during execution isn't verified by the Chains
+1. Verify that a TaskRun that's been modified after execution isn't verified by the Chains
+
+## Threat Model
+
+There are 5 main components in this threat model:
+- TektonController (Signer/Verifier)
+- Chains (Verifier)
+- TaskRun Pod/Entrypointer (Signer)
+- SPIRE Server (Authority)
+- SPIRE Agent (Workload Attestor)
+
+There are several capabilities that we want to ensure are correct:
+- Verification of signed Results are signed by the correct TaskRun Pod/Entrypointer step
+  - Verified by TektonController through SPIRE Authority
+- Verification of signed TaskRuns are signed by the TektonController
+  - Verified by TektonController and Chains through SPIRE Authority
+
+Due to the implementation of these different components being similar in nature of how they interact with the architecture (k8s, SPIRE, etc.), the two cases can be evaluated against the same threats.
+
+Threats vectors around verification fall into several categories:
+- Ability to create a verifiable false signature
+- Able to influence a skip of verification step
+- Able to influence verification to verify against a false authority
+
+### Ability to create a verifiable false signature
+
+Potential threats:
+- Access to SPIRE upstream authority lets an attacker sign arbitrary values.
+- Access to pod execution environment allows minting false signatures for that TaskRun step
+  - Through exec'ing into a pod (via api-server).
+  - Vertical attack from host.
```

Comment:
could you explain or link to something that shows what a vertical attack is for noobs like me? :)